### ISSUE TITLE
Import needs path

### ISSUE THREAD

**Issue by skaller** [vtereshkov/umka-lang#30]
I have a working Felix binding to basic Umka, here is the run:
```
~/felix>flx umkahello.flx
init ok = true
compile ok = true
Hello Umka!
run ok = true

~/felix>cat umkahello.flx
include "std/umka";
open Umka;

var amt = 1024 * 1024;
var umka = umkaAlloc();
var hellostr =
c"""fn main() {
    printf("Hello Umka!\\n")
}
""";

var ok = umkaInit (umka, c"Dummy", hellostr, amt, amt,0,C_hack::cast[+(+char)]0);
println$ "init ok = " + ok.str;
ok = umkaCompile umka;
println$ "compile ok = " + ok.str;
ok = umkaRun umka;
println$ "run ok = " + ok.str;
umkaFree umka;
```

This is fine! However to do anything complex I need to be able to import modules, in particular, the standard library `std.um`. This raises some issues:

First, loading a file from an embedded application is not acceptable, nor is loading it from an executable written in C and redistributed to clients: the executable should just run from anywhere and with no need to access the client file system. This is because Umka is intended as a small footprint micro embedded interpreter.

Secondly, even  with a hosted application, the file search algorithm is not specified. 

Third, using an operating system specific filename is not acceptable. Note that in C, using Unix filenames works on all platforms including Windows.

# Possible solutions

One way to fix this is to decouple the language specific module loading from its location.
Python does this, for example, and so does Felix. You could say:
```
require module "std";
```
in Umka. Now, how do we find it? The most general way is to use a C API call to provide a string containing the module text, and associate it with the module name, something like:
```
bool ok = umkaRegisterModule (umka, "modname", "//module code goes here");
```
Then, the C program is responsible for finding the file and loading it, or, it can just physically include the text of the module as a C string, which is what you'd want for an embedded application.

Note that in a real embedded application, the programmer would want strict control over what C functions Umka scripts written by *their* clients could call, and to pass any security vetting the host C program would necessarily have to be self-contained, and visibly restrict Umka scripts to a fixed, trusted, set of C functions.

Just an FYI: it's worse. The embedded module scripts would almost certainly *also* have to be encrypted either for commercial or national security reasons (string literals in executables are easy to not only see, but also modify). But this is another story.

**Comment by vtereshkov** [comment]
@skaller I'm aware of this problem. But I dislike the Python approach, since it leads to a total mess when several module versions exist in parallel. I think this is the reason why Go requires explicit file paths in the `import` clause, and so does Umka. The path in Umka is always relative to the module from which the file is imported.

An available workaround is to mechanically concatenate a string containing all the library stuff you need with a string containing the main script, but without the `*` export marks or `import` clauses.

**Comment by skaller** [comment]
I agree it's a workaround if you want to spend time editing all the library code.

I can cite an example: Felix parser actions are written in Scheme, using an embedded interpreter. Some code I wanted to use imports modules .. so I couldn't use it. You see, the problem wasn't file paths, since the Scheme is in files in the first place finding imports would be reliable and easy: the problem was dependencies. 

Most library code built from files is made of small units with a small number of dependencies but the transitive closure is not only huge and contain a lot of useless stuff, but invariable has some wart fixing crud in it which is "the one place we don't follow the standard" .. which breaks the whole thing.

I dislike ALL the solutions including (if not especially) the one I implemented in Felix myself. It's the same as shared library linkage problem all over again but for source code.

I actually think you are wise doing nothing about this until you have a solution you are happy with. The danger is that people will solve it themselves in C .. and all choose a different, incompatible method which will inhibit sharing code.

The problem I face is I cannot use import. The problem with your workaround is that the standard library is not stable because Umka is new, so i will have to edit it repeatedly and manually if I want to track the master repository.

BTW: your syntax is a lot better than Go.

**Comment by marekmaskarinec** [comment]
I think this would be really good. It's really tedious now, especially with the standard library. I really like the go way of doing things.

**Comment by vtereshkov** [comment]
Solved by implementing `umkaAddModule()`. Closing as duplicate of #43.
